File: OneDrive/Bureau/pi_rami/Scoring_Aide/src/main/java/tn/esprit/scoringaideservice/service/SatelliteService.java
package tn.esprit.scoringaideservice.service;

import tn.esprit.scoringaideservice.dto.*;

import java.util.List;

public interface SatelliteService {

    SatelliteIndexDTO getIndices(Long terrainId);

    List<SatelliteEvolutionDTO> getEvolution(Long terrainId);

    SatelliteBiomasseDTO getBiomasse(Long terrainId);
}

File: OneDrive/Bureau/pi_rami/agro-ui/src/app/core/services/satellite.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { SatelliteIndexDTO, SatelliteBiomasseDTO } from '../models/satellite.model';
import { EvolutionScoreDTO } from '../models/score.model';

@Injectable({
  providedIn: 'root'
})
export class SatelliteService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/satellite`;

  getNdvi(terrainId: number): Observable<SatelliteIndexDTO> {
    return this.http.get<SatelliteIndexDTO>(`${this.apiUrl}/ndvi/${terrainId}`);
  }

  getEvolution(terrainId: number): Observable<EvolutionScoreDTO[]> {
    return this.http.get<EvolutionScoreDTO[]>(`${this.apiUrl}/evolution/${terrainId}`);
  }

  getBiomasse(terrainId: number): Observable<SatelliteBiomasseDTO> {
    return this.http.get<SatelliteBiomasseDTO>(`${this.apiUrl}/biomasse/${terrainId}`);
  }
}
